data_structures/binary_tree.go

```diff
@@ -0,0 +1,35 @@
+package data_structures
+
+import (
+	"fmt"
+)
+
+type BtNode struct {
+	Data  int
+	Left  *BtNode
+	Right *BtNode
+}
+
+func (root *BtNode) PreOrderTraversal() {
+	if root != nil {
+		fmt.Printf("%d ", root.Data)
+		root.Left.PreOrderTraversal()
+		root.Right.PreOrderTraversal()
+	}
+}
+
+func (root *BtNode) InOrderTraversal() {
+	if root != nil {
+		root.Left.PreOrderTraversal()
+		fmt.Printf("%d ", root.Data)
+		root.Right.PreOrderTraversal()
```

Comment:
These should be `InOrderTraversal`

Reply:
Test out the following case:


        1
    2           3
4      5    6       7
    

Preorder -> 1 2 4 5 3 6 7
Inorder -> 4 2 5 1 6 3 7
Postorder -> 4 5 2 6 7 3 1

Preorder traversal is: 
1 2 4 5 3 6 7 
Postorder traversal is: 
4 5 2 6 7 3 1 
Inorder traversal is: 
4 2 5 1 6 3 7